main.cpp

```diff
@@ -244,33 +286,196 @@ IntType& IntType::divide ( const int y )
     return *this;
 }
 
+// Pow Implementations
+
+const FloatType& FloatType::pow ( float x )
+{
+    FloatType::powInternal( x );
+    return *this;
+}
+
+const FloatType& FloatType::pow ( FloatType& ft ) 
+{
+    FloatType::powInternal( static_cast<float>(ft) );
+    return *this;
+}
+
+const FloatType& FloatType::pow ( IntType& it ) 
+{
+    FloatType::powInternal( static_cast<float>(it) );
+    return *this;
+}
+
+const DoubleType& DoubleType::pow ( double x )
+{
+    DoubleType::powInternal( x );
+    return *this;
+}
+
+const DoubleType& DoubleType::pow ( DoubleType& dt )
+{
+    DoubleType::powInternal( static_cast<double>(dt) );
+    return *this;
+}
+
+const DoubleType& DoubleType::pow ( FloatType& ft )
+{
+    DoubleType::powInternal( static_cast<double>(ft) );
+    return *this;
+}
+
+const DoubleType& DoubleType::pow ( IntType& it )
+{
+    DoubleType::powInternal( static_cast<double>(it) );
+    return *this;
+}
+
+const IntType& IntType::pow ( int x )
+{
+    IntType::powInternal( x );
+    return *this;
+}
+
+const IntType& IntType::pow ( IntType& it ) 
+{
+    IntType::powInternal( static_cast<int>(it) );
+    return *this;
+}
+
+const IntType& IntType::pow ( FloatType& ft ) 
+{
+    IntType::powInternal( static_cast<int>(ft) );
+    return *this;
+}
+
+const IntType& IntType::pow ( DoubleType& dt ) 
+{
+    IntType::powInternal( static_cast<int>(dt) );
+    return *this;
+}
```

Comment:
None of these powInternal() require the fully qualified name to be used inside the function.
- [ ] remove clutter.